### PR TITLE
Refactor-Alert-CASE-1338

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
Added a function to accept alertMessage as an array and string

### CODE REVIEW

File: src/test/Alert_test.js [hunk 1]
```diff
@@ -1,25 +1,58 @@
...
+    />
   );
   const alert = {
     alertClassName: 'Success!',
     faIcon: 'fa-info-circle icon',
-    alertMessage: 'Error! you have to check back again!',
     alertCross: true,
   };
-  const comp = mount(<Alert />);
+  const comp = shallow(<Alert />);
```
Comment (edgibbs) [inline src/test/Alert_test.js]: Prefer component here over comp

File: src/test/Alert_test.js [hunk 2]
```diff
@@ -1,25 +1,58 @@
...
+      reasons: [],
+      messageData() {
+        if (Array.isArray(alertData.alertMessage)) {
+          return alertData.alertMessage.map(item => {
+            alertData.reasons.push(item.activation_reason_code);
+          });
+        } else return alertData.alertMessage;
+      },
+    };
+    const comp = shallow(<Alert />);
```
Comment (edgibbs) [inline src/test/Alert_test.js]: component versus comp

File: src/test/Alert_test.js [hunk 3]
```diff
@@ -1,25 +1,58 @@
...
+          activation_reason_code: 'Handguns in house',
+        },
+        {
+          activation_date: '4-5-2003',
+          activation_reason_code:
+            'Aggressive dog on premises, yard not fenced ',
+        },
+      ],
+      reasons: [],
+      messageData() {
```
Comment (edgibbs) [inline src/test/Alert_test.js]: I don't understand why this exists in the test.  There' already a default implementation of messageData() in the Alert component that should return items wrapped in a `<p>` tag or if not an array just returns a string.

File: src/test/Alert_test.js [hunk 4]
```diff
@@ -1,25 +1,58 @@
...
+          return alertData.alertMessage.map(item => {
+            alertData.reasons.push(item.activation_reason_code);
+          });
+        } else return alertData.alertMessage;
+      },
+    };
+    const comp = shallow(<Alert />);
+    comp.setProps(alertData);
+    it('displays array', () => {
+      expect(alertData.messageData().length).toEqual(2);
```
Comment (edgibbs) [inline src/test/Alert_test.js]: This isn't really testing the implementation in `<Alert/>`

File: package-lock.json [hunk 5]
```diff
@@ -1,9 +1,186 @@
 {
   "name": "react-wood-duck",
-  "version": "0.1.50",
+  "version": "0.1.51",
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+    "@babel/code-frame": {
```
Comment (eckdanny-osi) [inline package-lock.json]: why are `package-lock.json` _and_ `yarn.lock` lockfiles present. It only makes sense to have one or the other.

File: src/Alert.js [hunk 6]
```diff
@@ -4,15 +4,21 @@ import PropTypes from 'prop-types';
 const Alert = props => {
   const classField = `alert-message ${props.alertClassName}-message`;
   const faIcons = `fa ${props.faIcon}`;
-
+  const messageData = () => {
+    if (Array.isArray(props.alertMessage)) {
+      return props.alertMessage.map((item, i) => <p key={i}> {item}</p>);
```
Comment (eckdanny-osi) [inline src/Alert.js]: I don't think this is the design intent of `Alerts`... If there exists (for example) 4 messages, i'm pretty certain the desired visual representation if to show 4 alert instances (not a single alert with 4 paragraphs in it)

File: src/Alert.js [hunk 7]
```diff
@@ -27,7 +33,7 @@ const Alert = props => {
 Alert.propTypes = {
   alertClassName: PropTypes.string,
   faIcon: PropTypes.string,
-  alertMessage: PropTypes.string,
+  alertMessage: PropTypes.any,
```
Comment (eckdanny-osi) [inline src/Alert.js]: avoid explicit `any`s. If you _do_ mean to accept `String` or `Array<String>` use the `oneOfType` combinator

**Comment by eckdanny-osi** [comment]
Paired for a smidge with @Sushmareddy-OSI. As of right now CALs nor Intake have a dependency on `Alert` (or the public API (prop) `alertMessage`).

Balancing _need for change_ and _stability in the public API_ is absolutely a desired outcome for the component library maintenance planned for the immediate future.

**Pull request closed without merging**